app/subscribers/user_event_subscriber.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class UserEventSubscriber
+  def after_commit(event)
+    case event
+    when Users::UserDeletedEvent
+      SegmentDeleteUserWorker.perform_later(event.event_id)
```

Comment:
does it make sense to schedule a worker even if `Features::SegmentDeletionConfig.enabled?` is `false` ?

i mean, you do this check within the service, but wouldn't be better to the check up here?

Reply:
Makes sense, good catch 👍 💪 Done 😎 💃